### PR TITLE
CompletionConfig for realtime tables

### PR DESCRIPTION
**Pull request by npawar**
Based on discussion in https://github.com/apache/incubator-pinot/issues/4345
Tested the change on a test cluster

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/config/CompletionConfig.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * Class representing configurations related to realtime segment completion.
+ *
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CompletionConfig {
+
+  @ConfigKey(value = "completionMode")
+  private String _completionMode;
+
+  public String getCompletionMode() {
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/config/CompletionConfig.java]: Can we add some documentation here on what values can exist for completion mode? Please use @ ConfigDoc annotation giving examples as well.

File: pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java [hunk 2]
```diff
@@ -320,6 +320,11 @@ public ServerType getServerType() {
         IN_PROGRESS, DONE
       }
 
+      public enum CompletionMode {
+        DEFAULT, // default behavior
+        DOWNLOAD // always download the segment
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java]: please add more document the behavior here,as verbose as you can (comments are free :)). In the current implementation, we are planning on the server paying attention to this config, and deciding to download or build a realtime segment that it knows are equivalent either way.

**Comment by npawar** [comment]
> Can you also include the config for default starting numrows for segment size based threshold calcluations? We can then see how these two configs fit together
> 
> thanks

I added the comments. 
I would like to add the other config for default starting numrows in a future PR if that's okay (will send one out soon after this is merged)

File: pinot-common/src/main/java/org/apache/pinot/common/config/CompletionConfig.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Class representing configurations related to realtime segment completion.
+ *
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CompletionConfig {
+
+  @ConfigKey(value = "completionMode")
+  @ConfigDoc("Mode to use when completing segment. DEFAULT for default strategy (build segment if segment is equivalent to the committed segment, else download). DOWNLOAD for always download the segment, never build. ")
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/config/CompletionConfig.java]:
```suggestion
  @ConfigDoc(value = "Mode to use when completing segment. DEFAULT for default strategy (build segment if segment is equivalent to the committed segment, else download). DOWNLOAD for always download the segment, never build.", mandatory = false)
```

File: pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java [hunk 4]
```diff
@@ -320,6 +320,14 @@ public ServerType getServerType() {
         IN_PROGRESS, DONE
       }
 
+      /**
+       * Decides how the non-winner servers should get the segment
+       */
+      public enum CompletionMode {
+        DEFAULT, // default behavior - if the in memory segment in the non-winner segmis equivalent to the committed segment, then build and replace, else download
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java]: segmis?

File: pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java [hunk 5]
```diff
@@ -320,6 +320,14 @@ public ServerType getServerType() {
         IN_PROGRESS, DONE
       }
 
+      /**
+       * Decides how the non-winner servers should get the segment
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java]:
```suggestion
       * During realtime segment completion, the value of this enum decides how  non-winner servers should replace  the completed segment.
```

**Pull request merged**